cosmos/config.py

```diff
@@ -0,0 +1,210 @@
+            profiles_path = Path(self.path_to_profiles_yml)
+            if not profiles_path.exists():
+                raise ValueError(f"Could not find profiles.yml at {self.path_to_profiles_yml}")
+
+    def validate_profile(self) -> None:
+        "Validates that we have enough information to render a profile."
+        if not self.path_to_profiles_yml and not self.profile_mapping:
+            raise ValueError("Either path_to_profiles_yml or profile_mapping must be set to render a profile")
+
+    @contextlib.contextmanager
+    def ensure_profile(self, desired_profile_path: Path | None = None) -> Iterator[tuple[str, dict[str, str]]]:
+        "Context manager to ensure that there is a profile. If not, create one."
+        if self.path_to_profiles_yml:
+            yield self.path_to_profiles_yml, {}
+
+        elif self.profile_mapping:
+            profile_contents = self.profile_mapping.get_profile_file_contents(
+                profile_name=self.profile_name, target_name=self.target_name
+            )
+
+            if desired_profile_path:
+                logger.debug(
+                    "Writing profile to %s with the following contents:\n%s",
+                    desired_profile_path,
+                    profile_contents,
+                )
+                desired_profile_path.write_text(profile_contents)
+                yield str(desired_profile_path), self.profile_mapping.env_vars
+
+            else:
+                with tempfile.TemporaryDirectory() as temp_dir:
+                    temp_file = Path(temp_dir) / "profiles.yml"
+                    logger.debug(
+                        "Creating temporary profiles.yml at %s with the following contents:\n%s",
+                        temp_file,
+                        profile_contents,
+                    )
+                    temp_file.write_text(profile_contents)
+                    yield str(temp_file), self.profile_mapping.env_vars
+
+
+@dataclass
+class ExecutionConfig:
+    """
+    Contains configuration about how to execute dbt.
+
+    :param execution_mode: The execution mode for dbt. Defaults to local
+    :param dbt_executable_path: The path to the dbt executable. Defaults to dbt-ol or dbt if
+    available on the path.
+    :param dbt_cli_flags: A list of extra dbt cli flags to pass to dbt. Defaults to []
+    :param append_env: If True, append the env dictionary to the existing environment. If False,
+    replace the existing environment with the env dictionary. Defaults to False
+    :param cancel_query_on_kill: If True, cancel the query when the dbt process is killed. If False,
+    do not cancel the query when the dbt process is killed. Defaults to True
+    :param install_deps: If True, install dbt dependencies before running dbt. Defaults to False
+    :param skip_exit_code: If the dbt process exits with this exit code, do not raise an exception.
+    Defaults to None
+    """
+
+    execution_mode: str | ExecutionMode = ExecutionMode.LOCAL
```

Comment:
I really don't think we should support strings in our representation of `execution_mode`, `load_method`, and `test_behavior`, now that we are exposing these configs to end-users and breaking many Cosmos interfaces.

Using strings to represent constants/enums is error-prone - for both us and end-users.